code/modules/library/library_catalog.dm

```diff
@@ -0,0 +1,617 @@
+	var/list/forbidden_books = list(
+		/obj/item/book/manual/random,
+		/obj/item/book/manual/nuclear,
+		/obj/item/book/manual/wiki,
+		/obj/item/book/manual/hydroponics_pod_people,
+	)
+
+	var/newid = 1
+	//building a list of all programmatic books
+	for(var/typepath in (subtypesof(/obj/item/book/manual) - forbidden_books))
+		var/obj/item/book/B = new typepath(null)
+		var/datum/programmaticbook/PB = new()
+		PB.title = B.name
+		PB.author = B.author
+		PB.id = "M[newid]"
+		PB.path = typepath
+		newid++
+		books += PB
+
+/*
+ * can_vv_delete override
+ * Admins should not be deleting this willy nilly, if they think it is neccesary,
+ * they can go through the effort of advanced proccall
+ */
+/datum/library_catalog/can_vv_delete()
+	message_admins("An admin attempted to VV delete the global library catalog, this will break the library system for the round, if you know what you are doing please use advanced proccal")
+	return FALSE
+
+/*
+ * Database Select and Get Procs
+ *
+ * Each of these procs facilitate finding, taking,
+ * and prepping information from the database for use elsewhere
+ */
+
+///External proc that Returns a report library_category datum that matches the provided cat_id
+/datum/library_catalog/proc/getReportCategoryByID(category_id)
+	for (var/datum/library_category/category in GLOB.library_catalog.report_types)
+		if(category.category_id == category_id)
+			return category
+	//proc shouldn't get this far, but if there's an entry in the DB that we don't have added, just default to other cat
+	for (var/datum/library_category/category in GLOB.library_catalog.report_types)
+		if(category.category_id == LIB_REPORT_OTHER)
+			return category
+
+///External proc that Returns a book library_category datum that matches the provided cat_id
+/datum/library_catalog/proc/getBookCategoryByID(category_id)
+	for (var/datum/library_category/category in GLOB.library_catalog.categories)
+		if(category.category_id == category_id)
+			return category
+
+///External proc that Returns a report programmaticbook datum that matches the provided bookid
+/datum/library_catalog/proc/getProgrammaticBookByID(id)
+	for(var/book in GLOB.library_catalog.books)
+		var/datum/programmaticbook/PB = book
+		if(PB.id == id)
+			return PB
+
+/*
+  * # getBookByID
+  *
+  * External proc that takes in an id number and searches the Database for a book with a matching SSID
+  * returns a cached book with the data from that row
+  *
+  * Arguments:
+  * * id - integer value that matches as a book SSID
+ */
+/datum/library_catalog/proc/getBookByID(id)
+	var/datum/db_query/query = SSdbcore.NewQuery("SELECT id, author, title, content, summary, rating, raters, primary_category, secondary_category, tertiary_category, ckey, reports FROM library WHERE id=:id", list(
+		"id" = id
+	))
+
+	if(!query.warn_execute())
+		qdel(query)
+		return
+
+	var/list/results = list()
+	while(query.NextRow())
```

Comment:
Why use a `while` here when you return in the first loop?

Reply:
leftovers from some of the stuff I never cleaned out

Reply:
see AA's comment